docs/src/usage.md

```diff
@@ -0,0 +1,168 @@
+# Usage
+
+Distributed linear algebra frameworks are the backbone for efficient parallel
+codes in data analytics, scientific computing and machine learning. The central
+idea is that vectors and matrices can be partitioned into potentially
+overlapping chunks which are distributed across a set of workers on which we
+define the usual operations like products and norms.
+
+## Basic example
+
+In this section we take a look on solving the finite difference discretization
+of a Laplace problem in 1D over the domain [0,1]. As a reminder, the Laplace
+problem states to find function u(x) such that Δu(x) = 0 for all x ∈ [0,1].
+Without boundary conditions the problem is not well-posed, hence we introduce
+the Dirichlet condition u(0) = 1.
+
+Applying the finite difference method with length 0.25 we discretize the problem
+into linear system with 5 unkowns (u₁,...,u₅), which we call degrees of freedom:
+```math
+\frac{1}{4}
+\begin{pmatrix}
+1 &  0 &  0 &  0 &  0 \\
+0 & -2 &  1 &  0 &  0 \\
+0 &  1 & -2 &  1 &  0 \\
+0 &  0 &  1 & -2 &  1 \\
+0 &  0 &  0 &  1 & -1
+\end{pmatrix}
+\begin{pmatrix}
+u₁ \\
+u₂ \\
+u₃ \\
+u₄ \\
+u₅
+\end{pmatrix}
+=
+\begin{pmatrix}
+ 1 \\
+-1 \\
+ 0 \\
+ 0 \\
+ 0
+\end{pmatrix}
+```
+
+A detailed derivation can be found in standard numerical analysis lecture notes and books e.g. [these](https://people.sc.fsu.edu/~jburkardt/classes/math2071_2020/poisson_steady_1d/poisson_steady_1d.pdf). The linear system is then solved with
+conjugate gradients.
+
+### Commented Code
+
```

Comment:
Instead of writing a big box of code with coments in the source, I would write the example using cells like in a notebook and write the comments between the cells. This is possible with the `SequentialBackend()`. 

At the end you can collect all the cells in a code box and show how to use the MPI backend.

Reply:
Yes I also thought about introducing Literate.jl to make sure the minimal example is always executable, but I cannot find the time to cleanly introduce it.